edi_af_aisf_trask/models/res_partner.py

```diff
@@ -32,22 +34,43 @@ class ResPartner(models.Model):
     @api.one
     def write(self, vals):
         # Overwrite write method to catch changes to user_id
-        user_id = vals.get("user_id", False)
-        if user_id and user_id != self.env.user.id:
-            if user_id != self.user_id.id:
-                route = self.env.ref("edi_af_aisf_trask.asok_office_route")
-
-                msg_vals = {
-                    "name": "patch office msg",
-                    "edi_type": self.env.ref("edi_af_aisf_trask.asok_patch_office").id,
-                    "model": self._name,
-                    "res_id": self.id,
-                    "route_id": route.id,
-                    "route_type": "edi_af_aisf_trask_office",
-                }
-
-                message = self.env["edi.message"].create(msg_vals)
-                message.pack()
-                route.run()
+        if self.is_jobseeker:
+            user_id = vals.get("user_id", False)
+            request = self._get_request_object()
+            if request:
+                current_user = request.env.user.id
+            else:
+                current_user = self.env.user.id
+            if user_id and user_id != current_user or not (request and current_user == SUPERUSER_ID):
```

Comment:
This will evaluate to True if we don't have a request, which is wrong. There is no need to look at request at all. Just ensure that current_user != SUPERUSER_ID.